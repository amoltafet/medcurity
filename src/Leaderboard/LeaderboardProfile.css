.userPanel {
    margin-top: 1%;
    margin-left: 1%;
    margin-right: .5%;
    margin-bottom: .3%;
    background-color: #1e5b88;
    width: 98%;

}

@media screen and (min-width: 700px) {
    .userPanel {
        margin-top: .5%;
        margin-left: .5%;
        margin-right: .5%;
        margin-bottom: .2%;
        background-color: #1e5b88;
        width: 99%;
    }
}

.userRow {
    width: 100%;
    height: 100%;
}

.currentUserRow {
    width: 100%;
    height: 100%;
}

.companiesPanel {
    margin-top: 1%;
    margin-left: 1%;
    margin-right: .5%;
    margin-bottom: .3%;
    background-color: #127ebf;
    width: 98%;
}

@media screen and (min-width: 700px) {
    .companiesPanel {
        margin-top: .5%;
        margin-left: .5%;
        margin-right: .5%;
        margin-bottom: .2%;
        background-color: #127ebf;
        width: 99%;
    }
}


.displayLeaderboardInfo {
    width: 100%;
}

.leaderboardRank {
    font-family: Lato,sans-serif;
    font-size: 70%;
    font-weight: bold;
    color: white;
    margin-top: 10%;
}

@media screen and (min-width: 700px) {
    .leaderboardRank {
        font-family: Lato,sans-serif;
        font-size: 100%;
        font-weight: bold;
        color: white;
        margin-top: 8.5%;
    }
}

.userProfile {
    width: 60%;
    align-content: left;
}

@media screen and (min-width: 700px) {
    .userProfile {
        width: 50px;
        align-content: left;
    }
}

.userNameTitle {
    font-family: Lato,sans-serif;
    font-size: 70%;
    font-weight: bold; 
    color: white;
    margin-top: 5%;
    width: 100%;
    overflow:hidden !important;
    white-space: nowrap; /* Don't forget this one */
    text-overflow: ellipsis;
}

@media screen and (min-width: 700px) {
    .userNameTitle {
        font-family: Lato,sans-serif;
        font-size: 100%;
        font-weight: bold; 
        color: white;
        margin-top: 4.5%;
    }
}

.userPointsLeaderboard {
    font-family: Lato,sans-serif;
    font-size: 70%;
    color: white;
    margin-top: 3%;
}

@media screen and (min-width: 700px) {
    .userPointsLeaderboard {
        font-family: Lato,sans-serif;
        font-size: 100%;
        color: white;
        margin-top: 4%;
    }
}

.leaderboardBreak {
    width: 100%;
    height: .1rem;
    margin-bottom: 2%;
}

.category {
    font-family: Lato,sans-serif;
    font-size: 90%;
    text-align: center;
    color: white;
    margin-bottom: 5%;
}

.progressBar {
    width: 10vmin;
    margin: auto;
    margin-bottom: 10%;
    display: flex;
    flex: column;
    justify-content: center;
}

@media screen and (min-width: 700px) {
    .progressBar {
        width: 8vmin;
        margin: auto;
        margin-bottom: 10%;
        display: flex;
        flex: column;
        justify-content: center;
    }
}

.score_accordian_text {
    font-family: Lato,sans-serif;
    font-size: 90%;
    text-align: center;
    color: white;
    margin-bottom: 5% !important;
}

.accordian {
    border-color: transparent;
    background-color: transparent;
}

.accordianToggel {
    border-color: transparent;
    background-color: transparent;
    width: 100%;
}

.cardHeaderAccordian {
    background-color: transparent;
    border-color: transparent;
    width: 100%;
}

.cardAccordianCollapse {
    margin-left: .5%;
    margin-right: .5%;
    margin-bottom: 1.3%;
    background-color: #cc3333;
    width: 77%;
}

.categorgiesLeaderboardRow {
   margin-right: 1%;
   margin-left: 1%;
}