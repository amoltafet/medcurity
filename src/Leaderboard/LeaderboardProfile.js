import React from 'react';
import Accordion from 'react-bootstrap/Accordion'
import 'bootstrap/dist/css/bootstrap.min.css'
import { Row, Card, Image, Col } from 'react-bootstrap';
import { useEffect, useState } from "react";
import { CircularProgressbar } from 'react-circular-progressbar';
import axios from 'axios';
import './LeaderboardProfile.css'


/**
* Creates and displays each users leaderboard profile. 
* @param {Array} className the css style to display. 
* @param {user} user the user grabed from the dashboard.
* @return {GetPage}
*/
function LeaderboardProfile(props) {
    const [directories, setDirectories] = useState([]);
    axios.defaults.withCredentials = true;
    const [session, setSession] = useState([]);

    useEffect(() => {
        axios.get("http://localhost:3002/users/login").then((response) => {
          console.log('aaahhh', response.data.user)
          setSession(response.data.user[0])
        }).catch(error => console.error(`Error ${error}`));
      }, []);

    console.log(session)

    // Query for getting LearningDirectories Directory info
    useEffect(() => {
        axios.get('http://localhost:3002/api/getQuery', { params: { the_query: "SELECT * FROM LearningModulesDirectory" } }).then((response) => {
            setDirectories(Object.values(response.data))
        }).catch(error => console.error(`Error ${error}`));
    }, [])

    const GetCurrentModule = () => {
        var categoryData = []
        if (directories != null) {
        for (var i = 0; i < directories.length; i++) {
            console.log(directories[0])
            categoryData.push( <>
                <Row>
                    <Col >
                    <Card.Text className={props.className[2]}>{directories[i].Title}</Card.Text>
                    <CircularProgressbar
                        className={props.className[3]}
                        value={props.user.category1TotalPoints}
                        text={`${props.user.category1TotalPoints}%`}
                        styles={{
                            path: {
                                stroke: "#1e5b88",
                            },
                            trail: {
                                stroke: "#d8dae3",
                            },
                            text: {
                                fill: "white",
                                textAnchor: "middle",
                            }
                        }} />
                        </Col>
                        <Col>
                    <Card.Text className={props.className[2]}>Score</Card.Text>
                    <Card.Text className={props.className[2]}> 0</Card.Text>
                    </Col>
                </Row> 
               
            </>)
           
        } 
        return (categoryData)
        }
    }


    if (props.name === session.username) {
        return (
            <>
                <Card className={`uvs-left uvs-right ${props.className[0]}`} style={{ flexDirection: 'row' }}>
                    <Accordion className="displayLeaderboardInfo" defaultActiveKey="0">
                        <Accordion.Toggle eventKey="1" className="accordianToggel">
                            <Card className="cardHeaderAccordian" style={{ flexDirection: 'row' }}>
                                <Col sm>
                                    <div className="leaderboardRank">{props.index}.</div>
                                </Col>
                                <Col>
                                    <Image className={props.className[1]} src="/user.png" alt="" roundedCircle />
                                </Col>
                                <Col sm>
                                    <Card.Text className="userNameTitle">{props.name}</Card.Text>
                                </Col>
                                <Col sm>
                                    <div className="scoreLabelLeaderboard" >Total Score</div>
                                </Col>
                                <Col>
                                    <div className="userPointsLeaderboard">{props.user.overallPoints}</div>
                                </Col>

                            </Card>

                        </Accordion.Toggle>
                        <Accordion.Collapse className="accordianCollapse" eventKey="1">

                            <Row className="categorgiesLeaderboardRow">
                                {GetCurrentModule()}
                            </Row>

                        </Accordion.Collapse>
                    </Accordion>
                </Card>
            </>
        );
    }
    else {
        return (
            <>
                <Card className={`uvs-left uvs-right ${props.className[0]}`} style={{ flexDirection: 'row' }}>
                    <Accordion className="displayLeaderboardInfo" defaultActiveKey="0">
                        <Accordion.Toggle eventKey="1" className="accordianToggel">
                            <Card className="cardHeaderAccordian" style={{ flexDirection: 'row' }}>
                                <Col sm>
                                    <div className="leaderboardRank">{props.index}.</div>
                                </Col>
                                <Col>
                                    <Image className={props.className[1]} src="/user.png" alt="" roundedCircle />
                                </Col>
                                <Col sm>
                                    <Card.Text className="userNameTitle">{props.name}</Card.Text>
                                </Col>
                                <Col sm>
                                    <div className="scoreLabelLeaderboard" >Total Score</div>
                                </Col>
                                <Col>
                                    <div className="userPointsLeaderboard">{props.user.overallPoints}</div>
                                </Col>

                            </Card>

                        </Accordion.Toggle>
                    </Accordion>
                </Card>
            </>
        );
    }
}

export default LeaderboardProfile;