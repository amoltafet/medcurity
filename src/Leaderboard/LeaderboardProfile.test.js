import renderer from 'react-test-renderer';
import LeaderboardProfile from './LeaderboardProfile';
import React from 'react';

test('renders LeaderboardProfile', () => {
  const tree = renderer.create(<LeaderboardProfile />).toJSON();
  expect(tree).toMatchSnapshot();
});