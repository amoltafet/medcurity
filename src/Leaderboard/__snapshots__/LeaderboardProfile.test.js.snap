// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LeaderboardProfile 1`] = `
<div
  className="uvs-left uvs-right justify-content-center undefined card"
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <div
    className="displayLeaderboardInfo accordion"
  >
    <button
      className="accordianToggel"
      onClick={[Function]}
      type="button"
    >
      <div
        className="cardHeaderAccordian card"
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <div
          className="userRow text-center row"
        >
          <div
            className="col-lg-2 col-md-2 col-2"
          >
            <div
              className="leaderboardRank"
            >
              .
            </div>
          </div>
          <div
            className="col-lg-2 col-md-2 col-2"
          >
            <img
              alt=""
              className="rounded-circle"
              src="data:image/png;base64,"
            />
          </div>
          <div
            className="col-lg-3 col-md-3 col-4"
          >
            <p
              className="userNameTitle card-text"
            >
              <u />
            </p>
          </div>
          <div
            className="col-lg-4 col-md-5 col-4"
          >
            <div
              className="userPointsLeaderboard"
            >
              <b>
                Total Points:
              </b>
               
            </div>
          </div>
        </div>
      </div>
    </button>
    <div
      aria-expanded={null}
      className="accordianCollapse collapse"
    >
      <div
        className="categorgiesLeaderboardRow row row-cols-lg-6 row-cols-md-3 row-cols-3"
      >
        <img
          alt=""
          className="leaderboardBreak"
          src="/line.png"
        />
      </div>
    </div>
  </div>
</div>
`;
